### DISCUSSION TITLE
How are diagrams drawn

### DISCUSSION THREAD

**Discussion by tomas789** [oemof/tespy#430] [Q&A]
There are many slick diagrams all over the documentation. For example the one on top of this page. https://tespy.readthedocs.io/en/main/tutorials/heat_pump_steps.html#tespy-tutorial-heat-pump-label

In their source SVG code I can see they were generated by Inkscape but I cannot find the way how. 

How were those diagrams drawn? 

Thanks.

**Comment by fwitte** [comment]
@tomas789,

> There are many slick diagrams all over the documentation.

Thanks for that, appreciate it :).

> In their source SVG code I can see they were generated by Inkscape but I cannot find the way how.
>
> How were those diagrams drawn?

The uncomfortable truth is, that I have hand-drawn all of these with a template I created. Once you get a good flow with it, it is really quick and sometimes also a welcome change in work.

However, the programmer in me screams to implement automation based on the network structure or implement flowsheeting to generate the TESPy network based on the flow diagram. I am not really a good front-end developer, so I did not put a lot of time and effort into exploring the possibilities for this.

But there could be some options e.g.

- using draw.io: https://github.com/oemof/tespy/discussions/413 and it looks like there was some more work done based on that here https://github.com/jgraph/drawio/issues/3648.
- Other options might be using inkscape templates, for example something like https://github.com/upb-lea/Inkscape_electric_Symbols or,
- if you want to get real fancy have TeX generated figures, like https://github.com/tub-hofmann/chemplants-tub.

Best

Francesco
